### PR TITLE
[Feature Set] Fix ingest to raise error when user did not provide targets and called set_targets with_default=False

### PR DESCRIPTION
**Pull request by davesh0812**
[ML-3628](https://jira.iguazeng.com/browse/ML-3628)

### CODE REVIEW

File: mlrun/feature_store/api.py [hunk 1]
```diff
@@ -410,6 +410,18 @@ def ingest(
         raise mlrun.errors.MLRunInvalidArgumentError(
             "feature set and source must be specified"
         )
+    if not targets:
+        if not isinstance(featureset, FeatureSet):
+            raise mlrun.errors.MLRunInvalidArgumentError(
```
Comment (assaf758) [inline mlrun/feature_store/api.py:415]: i don't quite like this two separate checks for feature set uri vs FeatureSet.
but looking at set_targets() i think that if the targets parameter to ingest() and featureset.spec.targets are both empty/None we can raise (ie we don't need featureset.spec.with_default_targets). Do you agree?

File: mlrun/feature_store/api.py [hunk 2]
```diff
@@ -410,6 +410,18 @@ def ingest(
...
+                "targets must be specified\n"
+                "If you don't want to provide any target use `preview` function instead"
+            )
+        elif not (featureset.spec.targets or featureset.spec.with_default_targets):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"You disabled the default targets of the {featureset.metadata.name} feature set "
```
Comment (assaf758) [inline mlrun/feature_store/api.py:421]: please repharse without using "you". Something like:
no targets provided to feature set {featureset.metadata.name} ingest, aborting.\n
(`preview` can be used as an alternative to local ingest when targets are not needed)

"If you don't want to provide any target use preview function instead"

**Review by assaf758 — CHANGES_REQUESTED** [review]

File: mlrun/feature_store/api.py (hunk 1 above)
Comment (davesh0812) [reply mlrun/feature_store/api.py:415]: The question is what we want to do when the user didn't use set_targets function at all... I think in this case we need to use the default targets, and this is the reason I am using featureset.spec.with_default_targets.

Comment (davesh0812) [reply mlrun/feature_store/api.py:415]: Also if we get `featureset` from the use as uri (str) in this point it all-ready will be of  FeatureSet type.

**Review by assaf758 — APPROVED** [review]

**Pull request merged**